src/Xamarin.Android.Build.Tasks/Generator/Generator.cs

```diff
@@ -52,6 +53,13 @@ class Generator
 								message: xae.MessageWithoutCode,
 								messageArgs: new object [0]
 						);
+					} catch (DirectoryNotFoundException ex) {
```

Comment:
`DirectoryNotFoundException` is thrown on Windows when the path is too long?  Not `PathTooLongException`?  That seems weird. :-/

Reply:
Yeah they are translating the `Marshal.GetLastWin32Error()` code from the Windows APIs, and that is what it returns apparently...

The only cases I actually get `PathTooLongException` was if I the actual directory you try to delete is a long path. `Directory.DeleteDirectory` will throw `DirectoryNotFoundException` if it encounters a long path in a sub-directory. 🤷‍♂